.github/workflows/build.yml

```diff
@@ -0,0 +1,19 @@
+---
+# Basic build test
+
+name: build
+
+# yamllint disable-line rule:truthy
+on: [pull_request, push, workflow_dispatch]
+
+jobs:
+  build-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with: {fetch-depth: 50, submodules: recursive}
+
+      - name: build
+        run: cmake -B build/
+      - run: make -Werror -Wall -Wmissing-prototypes -Wimplicit-fallthrough=3
+             -Wpointer-arith -C build/
```

Comment:
Nobody uses CMAKE_C_FLAGS anymore unless this kind of operation is required.
Use [target_compile_options](https://cmake.org/cmake/help/latest/command/target_compile_options.html) to specify compilation flags per target. Especially now when we are going to interface with Zephyr. You probably do not want to enforce -Wall on Zephyr or other external dependency?

Reply:
You're confusing command line and scripts. Strange for someone who pretends to know how everyone in the world uses CMake (which is impossible considering how lax CMake is).

rimage is a build-time tool whereas Zephyr is cross-compiled. They have absolutely nothing related to each other.

But since you mentioned Zephyr, its CI uses EXTRA_CFLAGS command line interface to add `-Werror`. That does not seem standard.

Reply:
I do not like your tone @marc-hb nor share your opinion. Watch yourself we work in the company that promotes tolerance and mutual respect so please do so. I do not pretend anything. You pasted link to cmake flags that are considered bad practise. I do not understand why you are trying to add those flags here in the run script.
They are present anyway in rimage/CMakeLists.txt:
`target_compile_options(rimage PRIVATE
	-Wall -Werror -Wmissing-prototypes -Wimplicit-fallthrough
)`
Please explain to me why I am wrong skipping that misplaced attitude.

Reply:
@aborisovich this is not a script for users, its a CI test. anything here will not leak into other projects as nothing else is being built.

Reply:
@cujomalainey Ok. But why are those flags needed if we have them already defined in rimage/CMakeLists.txt?

Reply:
This was a copy paste conversion from what we had in the original Travis CI script, given this PR is almost 8months old and the travis script is probably from around 2 years ago, I would not be surprised if we no longer need them and they can be dropped.